projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-selection.service.ts

```diff
@@ -183,8 +183,8 @@ export class IgxTreeGridSelectionService extends IgxGridSelectionService {
      * populates the rowsToBeSelected and rowsToBeIndeterminate sets
      * with the rows which will be eventually in selected/indeterminate state
      */
-    private calculateRowsNewSelectionState(args: any) {
-        this.rowsToBeSelected = new Set<any>(args.oldSelection ? args.oldSelection : this.getSelectedRows());
+    private calculateRowsNewSelectionState(args: any, shouldConvert = false) {
+        this.rowsToBeSelected = new Set<any>(args.oldSelection ? shouldConvert ? args.oldSelection.map(r => r[this.grid.primaryKey]) : args.oldSelection : this.getSelectedRows());
```

Comment:
this.rowsToBeSelected will contain one undefined item when PK is not set on the grid